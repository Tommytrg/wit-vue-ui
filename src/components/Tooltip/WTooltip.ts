export enum TooltipPosition {
  CenterTop = 'center-top',
  RightTop = 'right-top',
  LeftTop = 'left-top',
  CenterBottom = 'center-bottom',
  RightBottom = 'right-bottom',
  LeftBottom = 'left-bottom'
}
export const tooltipPostions = Object.values(TooltipPosition)
