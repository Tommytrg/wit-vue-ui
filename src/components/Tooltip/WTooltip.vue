<template>
  <div class="group inline relative p-[4px] -p-[4px]">
    <slot name="main"></slot>
    <div
      class="tooltip-container hidden min-w-max py-md px-md rounded-lg group-hover:inline hover:inline transition-opacity text-sm rounded-md absolute bottom-md -translate-x-[50%] mb-sm"
    >
      <slot name="tooltip"></slot>
    </div>
  </div>
</template>

<script setup>
defineProps({
  bgColor: {
    type: String,
    required: true,
  },
})
</script>

<style scoped lang="scss">
.tooltip-container {
  @apply bg-[v-bind(bgColor)];
}
</style>
