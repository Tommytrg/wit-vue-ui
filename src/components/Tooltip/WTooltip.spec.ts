import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import WTooltip from './WTooltip.vue'

describe('WTooltip', () => {
  it('renders properly', () => {
    const wrapper = mount(WTooltip, {
      props: {
        bgColor: '#000'
      },
      slots: {
        main: `<p>Text</p>`,
        tooltip: `<p class="text-white-50">Text info</p>`
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
