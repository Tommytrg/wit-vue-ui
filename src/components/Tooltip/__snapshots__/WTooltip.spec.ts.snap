// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WTooltip > renders properly 1`] = `
<div
  class="group inline relative p-[4px] -p-[4px]"
  data-v-38c68585=""
>
  
  <p>
    Text
  </p>
  
  <div
    class="tooltip-container hidden min-w-max py-md px-md rounded-lg group-hover:inline hover:inline transition-opacity text-sm rounded-md absolute bottom-md -translate-x-[50%] mb-sm"
    data-v-38c68585=""
  >
    
    <p
      class="text-white-50"
    >
      Text info
    </p>
    
  </div>
</div>
`;
